src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1330 @@
+            auto rowHover = -1;
+
+            if (byte_525FC7 != 0xFF)
+            {
+                for (auto i = 0; i <= self->var_83C; i++)
+                {
+                    if (i >= self->var_83C)
+                    {
+                        if (byte_525FC7 == self->row_info[i])
+                        {
+                            rowHover = byte_525FC7;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        if (self->var_83C != 0)
+                        {
+                            rowHover = self->row_info[0];
+                        }
+                    }
+                }
+            }
+            else
+            {
+                if (self->var_83C != 0)
+                {
+                    rowHover = self->row_info[0];
+                }
+            }
+            self->row_hover = rowHover;
+            sub_458B51(self);
+            //sub_458BA0(self);     Ommitted as result not used
+        }
+
+        // 0x00457FFE
+        static void sub_457FFE(window* self)
+        {
+            self->min_width = new_industries::window_size.width;
+            self->min_height = new_industries::window_size.height;
+            self->max_width = new_industries::window_size.width;
+            self->max_height = new_industries::window_size.height;
+            input::toolSet(self, common::widx::tab_new_industry, 40);
+
+            input::set_flag(input::input_flags::flag6);
+            showGridlines();
+            byte_E0C3D9 = 0;
+            dword_E0C39C = 0x80000000;
+
+            self->var_83C = 0;
+            self->row_hover = -1;
+            self->var_846 = -1;
+
+            sub_458AAF(self);
+            //sub_458BA0(self);     Ommitted as result not used
+
+            _prng->rand_next();
+            _dword_E0C394 = _prng->srand_0();
+            _dword_E0C398 = _prng->srand_1();
+        }
+
+        // 0x00457F27
+        static void switchTab(window* self, widget_index widgetIndex)
+        {
+            if (input::is_tool_active(self->type, self->number))
+                input::cancel_tool();
+
+            self->current_tab = widgetIndex - widx::tab_industry_list;
+            self->frame_no = 0;
+            self->flags &= ~(window_flags::flag_16);
+
+            if (self->viewports[0] != nullptr)
+            {
+                self->viewports[0]->width = 0;
+                self->viewports[0] = nullptr;
+            }
+
+            auto tabInfo = tabInformationByTabOffset[widgetIndex - widx::tab_industry_list];
+
+            self->enabled_widgets = *tabInfo.enabledWidgets;
+            self->holdable_widgets = 0;
+            self->event_handlers = tabInfo.events;
+            self->activated_widgets = 0;
+            self->widgets = tabInfo.widgets;
+
+            if (self->current_tab == widx::tab_industry_list - widx::tab_industry_list)
+                sub_457FCA(self);
```

Comment:
we can give these better names. `industryListTabReset` or something same goes for L1258